openerp/addons/base/res/res_partner.py

```diff
@@ -546,6 +568,14 @@ def create(self, vals):
         # migrating to the new API
         c_type = vals.get('company_type', self._context.get('default_company_type'))
         is_company = vals.get('is_company', self._context.get('default_is_company'))
+        if not vals.get('image'):
+            parent_contact = self.env['res.partner'].browse(vals.get('parent_id'))
+            vals['image'] = vals.get('email') and self._get_gravatar_image(vals['email']) or parent_contact.image or self._get_default_image(is_company=vals.get('is_company'))
```

Comment:
@tivisse ,for invoice and delivery type address we have to set fixed image : which is "money.png" and "trunk.png"
and if we set email for contact address type , then we have to fetch from _get_gravatar_image and if email is not set then take parent image. correct me if i am wrong.